operator/hack/olm-operator-install.sh

```diff
@@ -16,34 +16,41 @@ function main() {
   esac
 
   local -r operator_ns="${1:-}"
-  local -r image_tag_base="${2:-}"
+  local -r index_image_repo="${2:-}"
 
   case $# in
   3)
-    local -r index_version="${3:-}"
-    local -r operator_version="${3:-}"
+    local -r index_image_tag="${3:-}"
+    local -r csv_version="${3:-}"
+    local -r operator_channel="latest"
     ;;
   4)
-    local -r index_version="${3:-}"
-    local -r operator_version="${4:-}"
+    local -r index_image_tag="${3:-}"
+    local -r csv_version="${4:-}"
+    local -r operator_channel="latest"
+    ;;
+  5)
+    local -r index_image_tag="${3:-}"
+    local -r csv_version="${4:-}"
+    local -r operator_channel="${5:-}"
     ;;
   *)
-    echo -e "Usage:\n\t$0 [--allow-dirty-tag | -d] <operator_ns> <image_tag_base> <index-version> [<install-version>]" >&2
-    echo -e "Example:\n\t$0 -d index-test quay.io/rhacs-eng/stackrox-operator 3.70.1 3.70.1" >&2
+    echo -e "Usage:\n\t$0 [--allow-dirty-tag | -d] <operator_ns> <index-image-repo> <index-image-tag> [<csv-version> [<install-channel>]]" >&2
+    echo -e "Example:\n\t$0 -d index-test quay.io/rhacs-eng/stackrox-operator-index v3.70.1 v3.70.0 latest" >&2
     echo -e "Note that KUTTL environment variable must be defined and point to a kuttl executable." >&2
     exit 1
     ;;
   esac
 
-  check_version_tag "${operator_version}" "${allow_dirty_tag}"
+  check_version_tag "${csv_version}" "${allow_dirty_tag}"
   create_namespace "${operator_ns}"
-  apply_operator_manifests "${operator_ns}" "${image_tag_base}" "${index_version}" "${operator_version}"
+  apply_operator_manifests "${operator_ns}" "${index_image_repo}" "${index_image_tag}" "${csv_version}" "${operator_channel}"
 
   if ! [[ "${USE_MIDSTREAM_IMAGES}" == "true" ]]; then
```

Comment:
I know you did not touch that, but `USE_MIDSTREAM_IMAGES` is redundant concept in the entire `operator/` directory. If we are to make things better, let's get rid of it and introduce more direct argument `INSTALL_PLAN_APPROVAL=(Manual|Automatic)`.

Anticipating the suggestion to do it in a later PR, my vote would be: no, let's try make it here (or in a stacked PR on top).

My further suggestion would be, in `tests/e2e/lib.sh`, to call `operator/hack/olm-operator-install.sh` directly instead of `make ... deploy-via-olm` in case of midstream images code branch.

https://github.com/stackrox/stackrox/blob/ed67f09fb494682790d623a5dc6863cf2cce141e/tests/e2e/lib.sh#L185-L194

The reason is that we override everything anyway. Why bother pushing things from the Makefile in a confusing way if we can call the thing directly. There will be a need to define `KUTTL` env variable but that's not critical given that we know the binary goes to a fixed path `operator/tools/kuttl`.

Reply:
Good idea about `$INSTALL_PLAN_APPROVAL`. Done.

However I'm quite strongly against calling anything in `operator/hack` from outside `operator/`. The makefile targets serve as the interface and give us a layer of indirection. For example I'd really prefer to avoid chasing uses of `KUTTL=` around if an when we migrate to chainsaw.

Reply:
Shell files calling Makefile calling shell files. I don't think this is a great design to stick to but if the kuttl thing is important for you, I would not insist.